lib/github_importer.rb

```diff
@@ -7,20 +7,45 @@ def initialize(oauth_token, repo)
     @repo = repo
   end
 
+  # Creates a HarvestedResource for each file in the directory
   # Yields to the block once for each file in the directory
-  # @yield [path, file] Gives the filename and filecontents to the block for
-  def import(path)
+  # @yield [HarvestedResource] Gives the harvested resource
+  def import(path, harvest, pipeline)
     gh.contents(repo, path: path).each do |resource|
-      yield(resource.path, retrieve_file(resource.path))
+      yield(retrieve_file(resource.path, harvest, pipeline))
     end
   end
 
   private
 
   attr_reader :gh, :repo
 
-  def retrieve_file(path)
+  # Creates a HarvestedResource and an ResourceContent if it doesn't already exist
+  # for the resource harvested from Github
+  # @param path [String] the resource to retrieve
+  # @param harvest [Harvest] the harvest instance this resource belongs to
+  # @param pipeline [Pipeline] the transformation pipeline this resource belongs to
+  # @return [HarvestedResource] the newly created resource from github
+  def retrieve_file(path, harvest, pipeline)
     blob = gh.contents(repo, path: path)
-    Base64.decode64(blob.content).force_encoding('UTF-8')
+    multihash = sha1_to_multihash(blob.sha)
+    resource = harvest.harvested_resources.create!(multihash: multihash,
+                                                   url: blob.git_url,
+                                                   original_filename: blob.path,
+                                                   pipeline: pipeline)
+    ResourceContent.persist(multihash) { decode(blob.content) }
+    resource
+  end
+
+  # @param encoded [String] a Base64 encoded String
+  # @return [String] the decoded text with UTF-8 encoding
+  def decode(encoded)
+    Base64.decode64(encoded).force_encoding('UTF-8')
+  end
+
+  def sha1_to_multihash(sha1)
```

Comment:
What about pulling in https://github.com/multiformats/ruby-multihash? It seems pretty light-weight, at least. Either that, or can we include more information about multihashes are, where the `fn` comes from, etc.